### DISCUSSION TITLE
Google provider for android and ios

### DISCUSSION THREAD

**Discussion by crabstars** [pocketbase/pocketbase#5726] [Q&A]
When reading the google oauth docs i found following: "You must create an OAuth client appropriate for the platform on which your app will run, ..." https://developers.google.com/identity/protocols/oauth2
But if i want to enable the google provider in pocketbase i have to provide a clientId and client secret. But I have two client, android and ios. Is there a way to enable it for both?

**Comment by ganigeorgiev** [comment]
You need to chose the first server-side option, aka. the "web" client type.

You can ignore this warning from the Google docs:
> _Do not use this client type for any other application, such as native or mobile apps._

It doesn't have any explanation why you shouldn't do it but there is nothing wrong or insecure with using the browser to perform the authentication and this is how it works with every other non-Google/Apple provider.

For a more "native" experience on mobile you can make use of [Android custom tabs](https://developer.chrome.com/docs/android/custom-tabs) to open the auth url (_for iOS the equivalent could be SafariViewController or other similar std service_). For Flutter I've tested [`flutter_custom_tabs`](https://pub.dev/packages/flutter_custom_tabs) in the past and it works fine (at least on Android).

**Reply by crabstars** [reply]
thanks for the clarification, i'll try it :)

**Reply by crabstars** [reply]
Hello, I am facing an issue where I don't have now a SHA-1 fingerprint. How can my app authenticate with the Pocketbase backend, ensuring that Pocketbase serve responses only to the correct app? @ganigeorgiev

**Reply by ganigeorgiev** [reply]
I have no idea what do you mean but please don't ping for no reason. I'm not a helpdesk.

My recommendation is to follow the guides in https://pocketbase.io/docs/authentication/#oauth2-integration. If that doesn't help then I'm not sure how to help further.

**Reply by markonadj1** [reply]
Hi @ganigeorgiev , thanks for sharing your setup

You mentioned using a single Google OAuth web client ID with both Android and iOS apps through PocketBase. I’m planning a similar architecture where both mobile apps use a system browser to authenticate via Keycloak, which in turn uses a single Google OAuth web client.

Just wondering — do you have this configuration running in production, or has it only been tested in development? Any challenges or limitations you’ve encountered would also be helpful to know.

Thanks in advance!
